### PR TITLE
fix(presets): support optional generator for configure presets >= v3

### LINKED ISSUE #2734: Different behavior between extension and CMake cli, when choosing the default generator in presets
### Brief Issue Summary

As the generator field is optional in a preset using CMake directly falls back to the default generator discovery procedure. I would expect the vscode extension to do the same... but it falls back to Ninja as generator, according to observed behavior and this code block: https://github.com/microsoft/vscode-cmake-tools/blob/01c4796fca985956fb19acda0e8e7ff7d0425391/src/preset.ts#L707-L712

From the CMake docs at https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html:
> An optional string representing the generator to use for the preset. If generator is not specified, it must be inherited from the inherits preset (unless this preset is hidden). In version 3 or above, this field may be omitted to fall back to regular generator discovery procedure.

I would like to know if there is a reason (I might have overlooked) for this behavior... if not I would be happy to create a PR :slightly_smiling_face: 

### CMake Tools Diagnostics

_No response_

### Debug Log

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by jochil**
Fixes #2734

### This changes behavior

The following changes are proposed:

- Adding a configure preset (>=v3) that has no generator as a test case
- Remove requirement and fallback for a generator if preset is >= v3

### CODE REVIEW

**Comment by jochil** [comment]
@bobbrow can you please approve the workflow run... I was not able to get `yarn run unitTests` working on my machine so it would be nice to know it there are some things waiting for a fix :)

**Review by bobbrow — COMMENTED** [review]
I meant to submit this review a few days ago.  There are a few more things for me to check, but I'm sending this now.

**Comment by bobbrow** [comment]
Some things are broken if `binaryDir` is not set.  I believe cmake defaults to use the cwd when a build directory is not supplied, but we currently set the cwd to `binaryDir` which doesn't have a value.  An update would need to be made to the extension to handle this properly.  I think it's easiest to just keep the code that assigns a default value for the `binaryDir`.

File: test/extension-tests/single-root-UI/project-folder/CMakeUserPresets.json [hunk 1]
```diff
@@ -19,19 +19,31 @@
...
+        "CMAKE_CXX_COMPILER": "cl.exe"
+      },
+      "environment": {
+        "TEST_VARIANT_ENV": "0cbfb6ae-f2ec-4017-8ded-89df8759c502"
+      }
+    },
+    {
+      "name": "NoGenerator",
+      "description": "Skips setting a generator",
+      "binaryDir": "${workspaceFolder}/build",
```
[the remark this answers is not in the document]
Comment (bobbrow) [reply test/extension-tests/single-root-UI/project-folder/CMakeUserPresets.json:38]: Correct. I'll resolve this conversation

**Review by bobbrow — APPROVED** [review]

**Comment by jochil** [comment]
@bobbrow sounds good! Feel free to go on :slightly_smiling_face:

**Review by elahehrashedi — APPROVED** [review]

**Comment by bobbrow** [comment]
For some reason the license/cla bot is not reporting that you signed the Contributor License Agreement.  But at the same time I see this:
image

We can push this PR through, but I wanted to double check with you that you agree to the CLA before doing so.

**Comment by jochil** [comment]
Yes, I agree to the CLA

I have no idea how that happened... I noticed it, but I was unable to sign again

**Pull request merged**